app/views.py

```diff
@@ -140,62 +148,122 @@ def createclass():
+            flash('User added')
+        else:
+            flash('User already in Class')
+        return redirect(url_for('forum', groupID=groupID))
+
 @app.route('/quiz/<int:quizID>')
+@login_required
 def preview_quiz(quizID):
-    """Renders the create class was a success page for educators."""
+    """Renders the preview quiz page for educators."""
     if not current_user.check_educator():
         return render_template('error404.html'), 404
     quiz = validate_quiz_link(quizID)
     questions = get_question_quiz(quiz)
-    return render_template('classquizzes.html', title=' | Create Class', questions=questions)
+    form = DeleteForm()
+    return render_template('previewquiz.html', title=' | Create Class', questions=questions, quiz=quiz, form=form)
 
-@app.route('/quiz', methods=['POST'])
+@app.route('/quiz/create', methods=['POST'])
+@login_required
 def createquiz():
     """Renders the create quiz page for educators."""
     if not current_user.check_educator():
         return render_template('error404.html'), 404
-    classForm = CreateName(prefix='class')
-    quizForm = CreateName(prefix='quiz')
+    classForm = NameForm(prefix='class')
+    quizForm = NameForm(prefix='quiz')
     image_file = url_for('static', filename='resources/images/profile_pics/' + current_user.image_file)
     if quizForm.validate_on_submit():
-        quiz = add_quiz(current_user, quizForm.name.data)
+        quiz = add_quiz(current_user, quizForm.title.data)
         return redirect(url_for('createqn', quizID=quiz.id))
     return render_template('dashboard.html', image_file=image_file, classForm=classForm, quizForm=quizForm)
 
 @app.route('/quiz/<int:quizID>/question', methods=['GET', 'POST'])
+@login_required
 def createqn(quizID):
     """Renders the add questions page for educators."""
     if not current_user.check_educator():
         return render_template('error404.html'), 404
     quiz = validate_quiz_link(quizID)
-    form = CreateQuestion()
+    form = QuestionForm()
     if form.validate_on_submit():
         #Commit inputs to database
         options = (form.op1.data, form.op2.data, form.op3.data, form.op4.data)
-        topic = get_topic(form.topic.data)
-        question = add_question(form.qn.data, options, int(form.corrOp.data), topic.id)
+        question = add_question(form.qn.data, options, form.corrOp.data, form.topic.data)
         add_question_quiz(quiz, question)
         if form.complete.data:
             return redirect(url_for('createquizsuccess', quizID=quizID))
         return redirect(url_for('createqn', quizID=quizID))
 
     return render_template('createqn.html', title=' | Create Quiz', form=form, quizID=quizID)
 
+@app.route('/quiz/<int:quizID>/question/<int:qnID>/delete', methods=['POST'])
+@login_required
+def deleteqn(quizID, qnID):
+    """Renders the add questions page for educators."""
+    if not current_user.check_educator():
+        return render_template('error404.html'), 404
```

Comment:
It is a 403 error, so 404 is for pages not found (removed/moved/deleted) and 403 is forbidden when users do not have permission to carry out certain actions.

Reply:
Use the suggestion feature.
```suggestion
        return render_template('error403.html'), 403
```